src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventBookingManager.java

```diff
@@ -926,49 +927,41 @@ private Long getPlacesAvailable(final IsaacEventPageDTO event, final boolean cou
     Map<BookingStatus, Map<Role, Long>> eventBookingStatusCounts =
         this.bookingPersistenceManager.getEventBookingStatusCounts(event.getId(), includeDeletedUsersInCounts);
 
-    long totalBooked = 0L;
-    Long studentCount = 0L;
-
-    if (eventBookingStatusCounts.get(BookingStatus.CONFIRMED) != null) {
-      for (Map.Entry<Role, Long> roleLongEntry : eventBookingStatusCounts.get(BookingStatus.CONFIRMED).entrySet()) {
-        if (Role.STUDENT.equals(roleLongEntry.getKey())) {
-          studentCount = roleLongEntry.getValue();
-        }
-
-        totalBooked = totalBooked + roleLongEntry.getValue();
-      }
+    Map<Role, Long> roleCounts;
+    if (countOnlyConfirmed) {
+      roleCounts = eventBookingStatusCounts.getOrDefault(BookingStatus.CONFIRMED, new HashMap<>());
+    } else {
+      roleCounts = getCombinedBookingCountsByRole(eventBookingStatusCounts);
     }
 
-    if (!countOnlyConfirmed && eventBookingStatusCounts.get(BookingStatus.WAITING_LIST) != null) {
-      for (Map.Entry<Role, Long> roleLongEntry : eventBookingStatusCounts.get(BookingStatus.WAITING_LIST).entrySet()) {
-        if (Role.STUDENT.equals(roleLongEntry.getKey())) {
-          studentCount = studentCount + roleLongEntry.getValue();
-        }
-
-        totalBooked = totalBooked + roleLongEntry.getValue();
-      }
+    // Calculate remaining capacity with a lower bound of zero (no negatives)
+    if (isStudentEvent) {
+      // For Student events we only limit the number of students, other roles do not count against the capacity
+      Long studentCount = roleCounts.getOrDefault(Role.STUDENT, 0L);
```

Comment:
Just curious why these are Longs rather than Integers?

Reply:
The previous version of the method used Long and I didn't think to change it. Not certain why UoC used Longs though . 🤷 
I'll take a look at where it's used but hopefully it should be fairly simple to change over? The value from the database query is read as Long currently but it's a COUNT rather than the field itself so I don't see why reading it as an Integer would have any issues...